app/src/test/java/com/x8bit/bitwarden/data/platform/repository/BridgeRepositoryTest.kt

```diff
@@ -0,0 +1,437 @@
+                userId = USER_2_ID,
+                decryptedUserKey = USER_2_UNLOCK_KEY,
+            )
+        } returns VaultUnlockResult.Success
+
+        // Setup settingDiskSource to have lastSyncTime set:
+        fakeSettingsDiskSource.storeLastSyncTime(USER_1_ID, LAST_SYNC_TIME)
+        fakeSettingsDiskSource.storeLastSyncTime(USER_2_ID, LAST_SYNC_TIME)
+
+        // Add some ciphers to vaultDiskSource for each user,
+        // and setup mock decryption for them:
+        every { vaultDiskSource.getCiphers(USER_1_ID) } returns flowOf(USER_1_CIPHERS)
+        every { vaultDiskSource.getCiphers(USER_2_ID) } returns flowOf(USER_2_CIPHERS)
+        mockkStatic(SyncResponseJson.Cipher::toEncryptedSdkCipher)
+        every {
+            USER_1_TOTP_CIPHER.toEncryptedSdkCipher()
+        } returns USER_1_ENCRYPTED_SDK_TOTP_CIPHER
+        every {
+            USER_2_TOTP_CIPHER.toEncryptedSdkCipher()
+        } returns USER_2_ENCRYPTED_SDK_TOTP_CIPHER
+        coEvery {
+            vaultSdkSource.decryptCipher(USER_1_ID, USER_1_ENCRYPTED_SDK_TOTP_CIPHER)
+        } returns USER_1_DECRYPTED_TOTP_CIPHER.asSuccess()
+        coEvery {
+            vaultSdkSource.decryptCipher(USER_2_ID, USER_2_ENCRYPTED_SDK_TOTP_CIPHER)
+        } returns USER_2_DECRYPTED_TOTP_CIPHER.asSuccess()
+    }
+
+    @AfterEach
+    fun teardown() {
+        confirmVerified(authRepository, vaultSdkSource, vaultRepository, vaultDiskSource)
+    }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `syncAccounts with user 1 vault unlocked and all data present should send expected shared accounts data`() =
+        runTest {
+            val sharedAccounts = bridgeRepository.getSharedAccounts()
+            assertEquals(
+                BOTH_ACCOUNT_SUCCESS,
+                sharedAccounts,
+            )
+            verify { authRepository.userStateFlow }
+            verify { vaultRepository.vaultUnlockDataStateFlow }
+            verify { vaultDiskSource.getCiphers(USER_1_ID) }
+            verify { vaultDiskSource.getCiphers(USER_2_ID) }
+            verify { vaultRepository.isVaultUnlocked(USER_1_ID) }
+            verify { vaultRepository.isVaultUnlocked(USER_2_ID) }
+            coVerify {
+                vaultRepository.unlockVaultWithDecryptedUserKey(
+                    userId = USER_2_ID,
+                    decryptedUserKey = USER_2_UNLOCK_KEY,
+                )
+            }
+            verify { vaultRepository.lockVault(USER_2_ID) }
+            coVerify { vaultSdkSource.decryptCipher(USER_1_ID, USER_1_ENCRYPTED_SDK_TOTP_CIPHER) }
+            coVerify { vaultSdkSource.decryptCipher(USER_2_ID, USER_2_ENCRYPTED_SDK_TOTP_CIPHER) }
+        }
+
+    @Test
+    fun `syncAccounts when userStateFlow is null should return an empty list`() = runTest {
+        every { authRepository.userStateFlow } returns MutableStateFlow(null)
+
+        val sharedData = bridgeRepository.getSharedAccounts()
+
+        assert(sharedData.accounts.isEmpty())
```

Comment:
Can we make this `assertTrue`

Reply:
Indubitably: 94c78b9a54bcb2e83518c9053b5c0137ede1c700